### ISSUE TITLE
stat_count does not work with line

### ISSUE THREAD

**Issue by dpmcsuss** [tidyverse/ggplot2#2274]
When I make an empirical probability mass functions (pmf), I like to use linerange with ymin=0 and ymax equal to the value of the empirical pmf. stat_count works well but I cannot use linerange with it without using kind of weird hack.

```r
library(ggplot2)
# works and gives a bar chart
ggplot(mtcars,aes(x=gear))+stat_count()

# throws an error
ggplot(mtcars,aes(x=gear))+stat_count(aes(ymax=..prop..,ymin=0),geom="linerange")
# Error: stat_count() must not be used with a y aesthetic.

# this works because it does not populate the ymin variable
ggplot(mtcars,aes(x=gear))+stat_count(aes(ymax=..prop..,ymin=..prop..-..prop..),geom="linerange")
```

The reason that it fails is because when I set `ymin=0` in `aes`, the data variable has a column with name `ymin` and `is.null(data$y)` returns true which results in the error. The hack involves setting `ymin=..prop..-..prop..` in the `aes` call.
https://github.com/tidyverse/ggplot2/blob/c592e3240beae5991a095aae06c02ce3fcfe571c/R/stat-count.r#L53

**Comment by has2k1** [comment]
How about?

```R
ggplot(mtcars,aes(x=gear))+stat_count(width=0.005)
```
For one, the desirable width will depend on the x scale. It is less wierd though.

**Comment by dpmcsuss** [comment]
Yes there are definitely some workarounds and not sure if this is really a bug but it is somewhat weird (though very `R`) behavior. To me, the `width` solution is less appealing than the`ymin=..prop..-..prop..` solution.

**Comment by hadley** [comment]
I think this will be fixed by #2059, since it would allow you to write:

```R
ggplot(mtcars, aes(gear)) + 
  stat_count(aes(ymax = computed(prop), ymin = 0, y = computed(0)), geom="linerange")
```

Otherwise, given that there are other ways to solve the problem (i.e. counting yourself) and it affects few people, I unfortunately don't have the time to make a fix.

**Comment by dpmcsuss** [comment]
Thanks. That looks like it will be a nice solution and I definitely understand that it is a minor issue.
